src/App.tsx

```diff
@@ -26,70 +26,134 @@ type ReorderOptions = {
+      visibleGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends Component<{}, State> {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleReverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  handleSortAlphabetically = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+    });
+  };
+
+  handleSortByLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+    });
+  };
+
+  handleReset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    const goods = getReorderedGoods(goodsFromServer, this.state);
+
+    const {
+      handleReverse,
+      handleSortAlphabetically,
+      handleSortByLength,
+      handleReset,
+    } = this;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={this.state.sortType === SortType.ALPHABET
+              ? 'button is-info '
+              : 'button is-info is-light'}
+            onClick={handleSortAlphabetically}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={this.state.sortType === SortType.LENGTH
+              ? 'button is-success '
+              : 'button is-success is-light'}
+            onClick={handleSortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={this.state.isReversed
+              ? 'button is-warning '
+              : 'button is-warning is-light'}
+            onClick={handleReverse}
+          >
+            Reverse
+          </button>
+
+          {(this.state.isReversed !== false
+          || this.state.sortType !== SortType.NONE)
```

Comment:
Move this to the variable
Also, you can sort it
also, make a destructuring from this.state
```suggestion
          {(this.state.isReversed
          || this.state.sortType !== SortType.NONE)
```